pkg/tsdb/elasticsearch/data_query.go

```diff
@@ -77,10 +77,17 @@ func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
 	e.logger.Info("Prepared request", "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
 	res, err := e.client.ExecuteMultisearch(req)
 	if err != nil {
-		// We are returning error containing the source that was added through errorsource.Middleware
+		if backend.IsDownstreamHTTPError(err) {
+			return errorsource.AddErrorToResponse(e.dataQueries[0].RefID, response, errorsource.DownstreamError(err, false)), nil
+		}
```

Comment:
Maybe slightly cleaner here to say 
```suggestion
		if backend.IsDownstreamHTTPError(err) {
			err = errorsource.DownstreamError(err, false)
		}
```

Reply:
Great suggestions! Thanks. Applied. 🙂